mongodump/mongodump.go

```diff
@@ -511,17 +526,24 @@ func (dump *MongoDump) DumpIntent(intent *intents.Intent, buffer resettableOutpu
 	if err != nil {
 		return err
 	}
-	findQuery := &db.DeferredQuery{Coll: session.Database(intent.DB).Collection(intent.C)}
+	intendedDB := session.Database(intent.DB)
+	coll := intendedDB.Collection(intent.C)
+	if dump.storageEngine == storageEngineUnknown {
+		if db.IsMMAPV1(intendedDB, intent.C) {
+			dump.storageEngine = storageEngineMMAPV1
+		} else {
+			dump.storageEngine = storageEngineModern
```

Comment:
Pls add a comment about how if we can't confirm MMAPv1 for any reason, we assume Modern.  That's probably in a doc comment in m-t-c but I think it would be helpful here as a reminder for a future reader.